Lib/ast.py

```diff
@@ -564,22 +568,40 @@ def __new__(cls, *args, **kwargs):
     type(...): 'Ellipsis',
 }
 
-class Index(AST):
+class slice(AST):
```

Comment:
@serhiy-storchaka what do you think about implementing custom instance and subclass checks for `slice`? `slice` is a sum type, so there should be checks like this `issubclass(thing, ast.slice)` or `isinstance(thing, ast.slice)`

something like this should work
```py
    def __subclasscheck__(self, subclass):
        if issubclass(subclass, Slice):
            return True
        else:
            return super().__subclasscheck__(subclass)

    def __instancecheck__(self, instance):
        if isinstance(instance, Slice):
            return True
        else:
            return super().__instancecheck__(instance)
```

Reply:
What is `Slice`?

Reply:
Before the simplification, it was part of `slice` kind. Now it is an expression by it is own but if we want to increase compatibility, wouldn't it better to it behave like a subclass of `ast.slice`?

Reply:
I do not think it would be useful because `Index()` can return arbitrary node type and it is impossible to distinguish `ExtSlice` from `Tuple`.